### DISCUSSION TITLE
using node-fetch within Gatsby Functions

### DISCUSSION THREAD

**Discussion by adiusz** [gatsbyjs/gatsby#31951] [Help]
Hello there,

Is it possible to use node-fetch within Gatsby Functions? I'm struggling to send a POST request to Slack's webhook, so I can send a message to a channel. 
Thing is I am using the same function in different project of mine, where I use Netlify Functions to do this. 
But here in Gatsby I got 500 error and console is logging `TypeError: fetch is not a function`

My function looks like that:
```
require('dotenv').config();
const fetch = require('node-fetch');

const handler = (req, res) => {
    const webbookUrl = process.env.SLACK_WEBHOOK_URL
    let webhookPayload = {
        "text": `My message`,
    }
    const webhookOptions = {
        method: 'POST',
        mode: 'no-cors',
        cache: 'no-cache',
        credentials: 'include',
        headers: {
            'Content-Type': 'application/json'
        },
        redirect: 'follow',
        referrerPolicy: 'no-referrer',
        body: JSON.stringify(webhookPayload)
    };

    fetch(webbookUrl, webhookOptions);
}
```

I would appreciate any help 🙃

**Comment by KyleAMathews** [comment]
First thing to double-check is did you install node-fetch? Also is there any other errors?

**Reply by adiusz** [reply]
yes node-fetch is installed, I triple-checked.
Also error is mentioning some gatsby-node file, have a look:
```
ERROR

fetch is not a function



  TypeError: fetch is not a function

  - processWarrantyRepairForm.js?:130 fnToExecute
    webpack://project/./src/api/processWarrantyRepairForm.js?:130:3

  - gatsby-node.ts:528
    [project]/[gatsby]/src/internal-plugins/functions/gatsby-node.ts:528:33


Executed function "/api/processWarrantyRepairForm" in 115ms

```

**Comment by smastrom-dolly** [comment]
Hi guys,

After an entire morning struggling to create a checkout with shopify-buy on server-side I finally sorted it out. On localhost everything was working but once deployed to Gatsby Cloud I was getting the same @adiusz error.
Installing and importing _node-fetch_ won't work, _isomorphic-fetch_ will give different errors but installing and importing **cross-fetch** will do the trick.

[https://www.npmjs.com/package/cross-fetch](url)
